plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -892,14 +899,54 @@ private String getProperty(Model model, String value) {
             throw new RuntimeException("UpdateSite.Data instantiation problems", e);
         }
     }
+    
+    /**
+     * Provides the Maven module used for a plugin on a `mvn [...] -pl` operation in the parent path 
+     * @param plugin
+     * @param pluginPath
+     * @return the maven module 
+     * @throws PomExecutionException 
+     * @throws IOException 
+     */
+    public static String getMavenModule(String plugin, File pluginPath, MavenRunner runner, MavenRunner.Config mavenConfig) throws PomExecutionException, IOException {
+        String absolutePath = pluginPath.getAbsolutePath();
+        if (absolutePath.endsWith(plugin)) {
+            return plugin;
+        }
+        String module = absolutePath.substring(absolutePath.lastIndexOf(File.separatorChar) + 1, absolutePath.length());
+        File parentFile = pluginPath.getParentFile();
+        if (parentFile == null) {
+            return null;
+        }
+        File log = new File(parentFile.getAbsolutePath() + File.separatorChar + "modules.log");
+        runner.run(mavenConfig, parentFile, log,"-Dexpression=project.modules", "-q", "-DforceStdout", "help:evaluate");
+        for (String line : org.apache.commons.io.FileUtils.readLines(log)) {
+            if (!StringUtils.startsWith(line.trim(), "<string>")) {
+                continue;
+            }
+            String mvnModule = line.replace("<string>", "").replace("</string>", "").trim();
```

Comment:
This could be a regexp match perhaps, but why are you attempting to parse XML this way? Use e.g. DOM4j.

Reply:
Yep, I will update, trying a quick-and-dirty approach just to test the entire solution

Reply:
Yes. Good enough for now I think.